render/filters.go

```diff
@@ -73,25 +73,24 @@ func ComposeFilterFuncs(fs ...FilterFunc) FilterFunc {
 // Filter removes nodes from a view based on a predicate.
 type Filter struct {
 	Renderer
-	FilterFunc   FilterFunc
-	filterPseudo bool
```

Comment:
Interesting idea. Will give it a go tomorrow morn.

A lot of these functions are only used in tests and can be moved there
making the surface area smaller.

On Wednesday, 15 June 2016, Alfonso Acosta <notifications@github.com> wrote:

> In render/filters.go
> <https://github.com/weaveworks/scope/pull/1581#discussion_r67230674>:
>
> > @@ -73,25 +73,24 @@ func ComposeFilterFuncs(fs ...FilterFunc) FilterFunc {
> >  // Filter removes nodes from a view based on a predicate.
> >  type Filter struct {
> >  	Renderer
> > -	FilterFunc   FilterFunc
> > -	filterPseudo bool
>
> My idea was to add an And combinator (like Complement) and combine it
> with IsNotPseudo in the APITopologyOptions, instead of embedding
> IgnorePseudo in the filter definitions.
>
> Something like
>
> func And(f FilterFunc, g FilterFunc) FilterFunc {
>      return func(node report.Node) bool { return f(node) && g(node)}
> }
> option :=  APITopologyOption{"stopped", "Stopped containers", render.And(render.IsStopped, render.IsNotPseudo)}
>
> Would that make the code simpler? I think that could make it cleaner but
> feel free to disregard it if you think it's too complex.
>
> —
> You are receiving this because you were assigned.
> Reply to this email directly, view it on GitHub
> <https://github.com/weaveworks/scope/pull/1581/files/f5913ec0e4c53e39daeb8f6ac20aac9020e13da4..43fecbec4760b2fca92dfb7418017ede997ae281#r67230674>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AAbGhYlU09Ot777VjXbH8SOYC6G8o3hnks5qMFRogaJpZM4I1NXa>
> .
>